baybe/parameters/categorical.py

```diff
@@ -47,9 +48,13 @@ def comp_df(self) -> pd.DataFrame:  # noqa: D102
         # See base class.
         if self.encoding is CategoricalEncoding.OHE:
             cols = [f"{self.name}_{val}" for val in self.values]
-            comp_df = pd.DataFrame(np.eye(len(self.values), dtype=int), columns=cols)
+            comp_df = pd.DataFrame(
```

Comment:
So I guess this is the part you mentioned in the chat, right? Seems reasonable, but I'm also not sure where is the best place for the cast. In principle, one could also use bool/int type here and then only cast when constructing the searchspace comp_df (because there, there whole array must share a single dtype), but I'm not sure if anything is gained by that...

Reply:
hmm id rather have specified types as close to the "source" as possible, imo it would be strange if `comp_df` had other types than the resulting `comp_rep`
Not sure if the `comp_df` is used anywhere else but issues could be coming up i we shift the type responsibility to the product utility

Reply:
Yeah, probably too much of a hassle. Happy to keep it as is 👍🏼